workspace_integration_test.go

```diff
@@ -277,6 +278,49 @@ func TestWorkspacesCreate(t *testing.T) {
 		assert.Nil(t, w)
 		assert.Error(t, err)
 	})
+
+	t.Run("when options include trigger-patterns (behind a feature flag)", func(t *testing.T) {
+		// Remove the below organization creation and use the one from the outer scope once the feature flag is removed
+		orgTest, orgTestCleanup := createOrganizationWithOptions(t, client, OrganizationCreateOptions{
+			Name:  String("tst-" + randomString(t)[0:20] + "-ff-on"),
+			Email: String(fmt.Sprintf("%s@tfe.local", randomString(t))),
+		})
+		defer orgTestCleanup()
+
+		options := WorkspaceCreateOptions{
+			Name:                String("foobar"),
+			FileTriggersEnabled: Bool(true),
+			TriggerPatterns:     []string{"/module-1/**/*", "/**/networking/*"},
+		}
+		w, err := client.Workspaces.Create(ctx, orgTest.Name, options)
+
+		require.NoError(t, err)
+		assert.Equal(t, options.TriggerPatterns, w.TriggerPatterns)
+
+		// Get a refreshed view from the API.
+		refreshed, err := client.Workspaces.Read(ctx, orgTest.Name, *options.Name)
+		require.NoError(t, err)
+
+		for _, item := range []*Workspace{
+			w,
+			refreshed,
+		} {
+			assert.Equal(t, options.TriggerPatterns, item.TriggerPatterns)
+		}
+	})
+
+	t.Run("when options include both trigger-patterns and trigger-paths error is returned", func(t *testing.T) {
+		options := WorkspaceCreateOptions{
+			Name:                String("foobar"),
+			FileTriggersEnabled: Bool(true),
+			TriggerPrefixes:     []string{"/module-1", "/module-2"},
+			TriggerPatterns:     []string{"/module-1/**/*", "/**/networking/*"},
+		}
+		w, err := client.Workspaces.Create(ctx, orgTest.Name, options)
+
+		assert.Nil(t, w)
+		assert.EqualError(t, err, ErrUnsupportedBothTriggerPatternsAndPrefixes.Error())
```

Comment:
Could we add another few lines below here to test the `WorkspaceUpdateOptions.valid()` case?